### PR TITLE
AssignByValueMakeLocalCopyQuickFix - corrects error when applied on a function 

### PR DESCRIPTION
**Pull request by BZngr**
The AssignByValueMakeLocalCopyQuickFix does not properly handle rewriting the AsTypeContext of a function.  This PR corrects the behavior.  

The current result when this QuickFix is applied to a Function:

Original code:
```vba
Private Function AddTwo(ByVal arg As Long) As Long
  arg = arg + 2
  AddTwo = arg
End Function
```

After the QuickFix makes a local copy of `arg` (Uncompilable):
```vba
Private Function AddTwo(ByVal arg As Long)
  Dim localArg As Long
  localArg = arg As Long  <-- Function's AsTypeContext is pushed to here
  localArg = localArg + 2
  AddTwo = localArg
End Function
```

The PR also enhances the QuickFix to retain any existing comment after the Sub/FunctionStmtContext and match the indentation format of the original code's first line.

### CODE REVIEW

File: Rubberduck.Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs [hunk 1]
```diff
@@ -13,6 +13,8 @@
 using Rubberduck.Parsing.Inspections.Resources;
 using Rubberduck.Parsing.Rewriter;
 using Rubberduck.Parsing.VBA;
+using static Rubberduck.Parsing.Grammar.VBAParser;
+using System.Diagnostics;
```
Comment (Vogel612) [inline Rubberduck.Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs]: not sure why that is included?

Comment (BZngr) [reply Rubberduck.Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs]: `Rubberduck.Parsing.Grammar.VBAParser` supports use/compiling of references to `EndOfStatementContext`, `AgListContext`, and `AsTypeClauseContext` .  There's a Debug.Assert statement used (which could be removed if needed)  - so `System.Diagnostics`.  Is there a better, more preferred approach?

Comment (Vogel612) [reply Rubberduck.Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs]: explicitly checking for your precondition.

Comment (BZngr) [reply Rubberduck.Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs]: @Vogel612 - TBH I'm not sure I understand your last comment.  Made some changes consistent with what I could find on the web after searching for "checking preconditions".  Hopefully the latest change is something close to what you are recommending.

**Pull request merged**